### ISSUE TITLE
Epic: Move ComputeUnit source of truth from scheduler to autoscaler-agent

### ISSUE THREAD

**Issue by sharnoff** [neondatabase/autoscaling#706]
## Problem description / Motivation

The autoscaler-agent makes scaling decisions. It's weird that it needs to successfully communicate with the scheduler to be able to decide what it wants ­— it only knows the compute unit (and therefore, how much to scale) once it successfully receives a `PluginResponse`.

Also, moving this from the scheduler to the autoscaler-agent unblocks #615.

## Feature idea(s) / DoD

ComputeUnit is defined in the autoscaler-agent's config, and nowhere else.

## Tasks

```[tasklist]
# Tasks
- [ ] #707
- [ ] #743
```

```[tasklist]
# Follow-ups
- [ ] https://github.com/neondatabase/autoscaling/pull/744
```

**Comment by shayanh** [comment]
Could you a little bit elaborate on this part?

> It's weird that it needs to successfully communicate with the scheduler to be able to decide what it wants

**Comment by sharnoff** [comment]
> Could you a little bit elaborate on this part?

Sure — currently, because the compute unit is defined by the scheduler and the autoscaler-agent only knows the compute unit when it receives a response from the plugin, it's unable to decide how to scale the VM until after it's communicated with the scheduler plugin.

This also means that _in theory_ the compute unit could change between requests, which is actually the reason why this is required for #680 (otherwise we'd lose the guarantees around vm-monitor requests that #615 will provide)

**Comment by sharnoff** [comment]
status: Waiting on review, low priority.

**Comment by sharnoff** [comment]
status: Unchanged.

**Comment by sharnoff** [comment]
status: Final PR has 1 approval, [just commented](https://github.com/neondatabase/autoscaling/pull/743#issuecomment-1951617959) asking if I should hold off merging to give a chance for another review.
